pytorch_toolkit/human_pose_estimation/datasets/transformations.py

```diff
@@ -284,47 +284,48 @@ def __call__(self, sample):
 
 class SinglePersonFlip(object):
     def __init__(self, prob=0.5):
-        super().__init__()
         self._prob = prob
+        self._left_keypoints_indice = None
+        self._right_keypoints_indice = None
 
     def __call__(self, sample):
-        prob = random.random()
-        do_flip = prob <= self._prob
+        rand = random.random()
+        do_flip = rand <= self._prob
         if not do_flip:
             return sample
 
         sample['image'] = cv2.flip(sample['image'], 1)
-
         w, h = sample['image'].shape[1], sample['image'].shape[0]
+        sample['center'] = w - sample['center'] - 1
         for id in range(len(sample['keypoints']) // 3):
             if sample['keypoints'][id * 3] == -1:
                 continue
             sample['keypoints'][id * 3] = w - 1 - sample['keypoints'][id * 3]
-        sample['keypoints'] = self._swap_left_right(sample['keypoints'])
+        self._swap_left_right(sample['keypoints'])
 
         return sample
 
     def _swap_left_right(self, keypoints):
-        right = [0, 1, 2, 3, 4, 5, 6, 7, 8, 30, 31, 32, 33, 34, 35, 36, 37, 38]
-        left = [15, 16, 17, 12, 13, 14, 9, 10, 11, 45, 46, 47, 42, 43, 44, 39, 40, 41]
-        for r, l in zip(right, left):
-            keypoints[r], keypoints[l] = keypoints[l], keypoints[r]
-        return keypoints
+        keypoints[self._right_keypoints_indice + self._left_keypoints_indice] =\
+            keypoints.copy()[self._left_keypoints_indice + self._right_keypoints_indice]
 
 
 class ChannelPermutation(object):
     def __init__(self, prob=0.5):
-        super().__init__()
         self._prob = prob
 
     def __call__(self, sample):
-        prob = random.random()
-        if prob > 0.5:
-            new_order = np.random.permutation(3)
-            image = sample['image']
-            image[:, :, 0], image[:, :, 1], image[:, :, 2] =\
-                image[:, :, new_order[0]], image[:, :, new_order[1]], image[:, :, new_order[2]]
-            sample['image'] = image
+        rand = random.random()
+        do_cp =rand > self._prob
+        if not do_cp:
```

Comment:
`if random.random() < self._prob: `

Should be `<` instead of `>`
Prob `0.9` means that transformation should be done if random value is from `[0,0.9]` interval